src/core/prometheus.cc

```diff
@@ -813,13 +840,16 @@ class metrics_handler : public httpd::handler_base  {
                     return do_with(get_range(families, metric_family_name, prefix),
                             [&s, this, is_text_format, show_help, filter](metric_family_range& m) {
                         return (is_text_format) ? write_text_representation(s, _ctx, m, show_help, filter) :
-                                write_protobuf_representation(s, _ctx, m);
+                                write_protobuf_representation(s, _ctx, m, filter);
                     });
+                }).handle_exception([] (auto ep) mutable {
+                    seastar_logger.warn("exception in hande: Failed to write protobuf metrics {}", ep);
                 }).finally([&s] () mutable {
```

Comment:
Seems unrelated. Also, use std::exception_ptr for the type.

Reply:
ok